query/language/fetch.feature

```diff
@@ -11,855 +11,2942 @@ Feature: TypeQL Fetch Query
+      """
+      match
+      $p isa! $t, has person-name $n;
+      fetch {
+        "first": $t,
+        "first": $n,
+      };
+      """
+    Then typeql read query; fails with a message containing: "multiple mappings for one key"
+      """
+      match
+      $p isa! $t, has person-name $n;
+      fetch {
+        "first": $t,
+        "first": [ $p.ref ],
+      };
+      """
+    Then typeql read query; fails with a message containing: "multiple mappings for one key"
+      """
+      match
+      $p isa! $t, has person-name $n;
+      fetch {
+        "first": $n,
+        "first": [ $p.person-name ],
+      };
+      """
+    Then typeql read query; fails with a message containing: "multiple mappings for one key"
+      """
+      match
+      $p isa! $t, has person-name $n;
+      fetch {
+        "first": $n,
+        "first": $n,
+      };
+      """
+
+
+  Scenario: same fetch parameter names in parent and sub fetches are permitted
+    When get answers of typeql read query
+      """
+      match
+      $p isa! $t, has person-name $n;
+      fetch {
+        "first": $t,
+        "second": [
+          match
+            $v = $n;
+          fetch {
+            "first": $n,
+            "second": $v
+          };
+        ],
+        "third": [
+          match
+            $p has age $v;
+          fetch {
+            "first": $v,
+            "second": $n,
+            "third": [
+              match
+                $f links (friend: $p, friend: $friend);
+                not { $friend has person-name "Alice"; };
+              fetch {
+                "first": [ $friend.person-name ]
+              };
+            ]
+          };
+        ],
+        "fourth": $n
+      };
+      """
+    Then answer contains document:
+      """
+      {
+        "first": "person",
+        "second": [
+          {
+            "first": "Alice",
+            "second": "Alice"
+          }
+        ],
+        "third": [
+          {
+            "first": 10.0,
+            "second": "Alice",
+            "third": [
+              {
+                "first": [ "Bob" ]
+              }
+            ]
+          }
+        ],
+        "fourth": "Alice"
+      }
+      """
+
+
+  Scenario: fetch subqueries can contain complex streams
+    Then get answers of typeql read query
+      """
+      match
+      $p isa! $t, has person-name $n;
+      fetch {
+        "subquery": [
+          match
+            $p has person-name $pn;
+            $nv = $n;
+            $pnv = $pn;
+            not { $nv == $pnv; };
+          sort $nv;
+          select $pnv;
+          limit 10;
+          fetch {
+            "another person name": $pnv
+          };
+        ]
+      };
+      """
+    Then answer contains document:
+      """
+      {
+        "subquery": [
+          {
+            "another person name": "Alice"
+          }
+        ]
+      }
+      """
+
+
+  Scenario: non-fetch subqueries are not permitted
```

Comment:
Similarly here - you know that all queries are either Row pipelines or Document-producing pipelines. You can probably just take 1 example of a Row-producing pipeline

Reply:
I want to make sure that users don't get their server crashed.